### PR TITLE
Adding docs to add new flags

### PR DESCRIPTION
**Pull request by komish**
Part of #1. I'm not sure what additional docs you would like. Docs on how to run it seem to be covered in README.md so I've just added a quick doc to have users add more flags if they should like. Happy to hear more ideas!

### CODE REVIEW

File: docs/adding-flags.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# Adding Flags
+
+Adding a flag to **yctf** is easy. While the project is written using the [gin-gonic/gin](https://github.com/gin-gonic/gin) framework, you do not need extensive knowledge using the framework to add a flag. That said, some knowledge is beneficial for you to be able to create unique flags in your handlers. 
+
+Each flag will require five additions to the cod base.
```
Comment (tonyskapunk) [inline docs/adding-flags.md]:
```suggestion
Each flag will require five additions to the code base.
```

File: docs/adding-flags.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+## Adding Metadatda
+
+Each flag has an entry in the `flags.json` file in the following format.
```
Comment (tonyskapunk) [inline docs/adding-flags.md]:
```suggestion
Each flag has an entry in the [`flags.json`](../flags.json) file in the following format.
```

File: docs/adding-flags.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+       "ID": 0,
+       "Flag": "yctf-5f39410913126cea50a3e6fcdb7312628a869da376985fdb7ef7d0a41254a0e3", 
+       "Template": "flag0.html"
+    }
+    ...
+]
+```
+
+- `ID` is the flag identifier. This must be unique.
+- `Flag` is a random string with the `yctf-` prefix that is to be found by the participant. This must be unique.
```
Comment (tonyskapunk) [inline docs/adding-flags.md]:
```suggestion
- `Flag` is a sha-2 (with a digest of 256) string with the `yctf-` prefix that is to be found by the participant. This must be unique. The sha-2 string is the hash of `yctf-ID`.
```

File: docs/adding-flags.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+- `Template` refers to a Go HTML Template by filename. These are stored in the `templates/` directory. This should closely mirror the format of existing templates (e.g. `flag{ID}.html`).
```
Comment (tonyskapunk) [inline docs/adding-flags.md]:
```suggestion
- `Template` refers to a Go HTML Template by filename. These are stored in the [`templates/`](../templates) directory. This should closely mirror the format of existing templates (e.g. `flag{ID}.html`).
```

File: docs/adding-flags.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+You will need to add an entry to `flags.json` for each flag that you want to introduced to `yctf`.
```
Comment (tonyskapunk) [inline docs/adding-flags.md]:
```suggestion
You will need to add an entry to [`flags.json`](../flags.json) for each flag that you want to introduce to `yctf`.
```

File: docs/adding-flags.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+	"fmt"
+	"log"
+
+	"github.com/gin-gonic/gin"
+)
+
+func getFlag99(c *gin.Context) {
+    // Get the flag details. These are pulled from the flags.json file.
+    f, err := getFlag(99)
+    // If we were unable to get the flag from teh json file, we return an error
```
Comment (tonyskapunk) [inline docs/adding-flags.md]:
```suggestion
    // If we were unable to get the flag from the json file, we return an error
```

File: docs/adding-flags.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+		"title":   title}, f.Template)
+}
+```
+
+You may use other approaches to rendering so long as you handle rendering of your flag's clue (which is always HTML). 
+
+
+## Adding Your Template
+
+The template is the next piece you need. It should live in the [templates/](../templates/) directory and should take on the same name you defined for your flag in `flags.json`.
```
Comment (tonyskapunk) [inline docs/adding-flags.md]:
```suggestion
The template is the next piece you need. It should live in the [templates/](../templates/) directory and should take on the same name you defined for your flag in [`flags.json`](../flags.json).
```

**Review by tonyskapunk — CHANGES_REQUESTED** [review]
Hi @komish this is great, thanks for taking the time to contribute to this project.

It's great to see that you nailed most of the things in here. I just added a bit of clarity to the Flag string and how it is generated.

Let me know if there is any comment on my suggestions or if there is any other question.

Thanks again!

File: docs/adding-flags.md (hunk 3 above)
Comment (komish) [reply docs/adding-flags.md]: Can you clarify the last piece?
> The sha-2 string is the has of `yctf-ID`.

Would it make sense to include a small pkg that can be run to spit out a hash on an input? Maybe using `go run pkg/generateflag/ ${id}`?

Either way, I've changed this wording.

**Comment by komish** [comment]
@tonyskapunk all should be corrected in the latest commit.

File: docs/adding-flags.md (hunk 3 above)
Comment (tonyskapunk) [reply docs/adding-flags.md]: Adding a function/pkg for that is part of #4 feel free to tackle that one too :smile_cat:

**Review by tonyskapunk — APPROVED** [review]
Everythin looks good, thanks for taking care of this @komish !

**Comment by tonyskapunk** [comment]
@all-contributors please add @komish  for documentation

**Pull request merged**